backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/VerkuendungsImportService.java

```diff
@@ -47,4 +88,200 @@ public UUID storeNormendokumentationspaket(Query query) throws IOException {
+    validateFileIsZipArchive(zipFile);
+    log.debug("File is zip archive.");
+
+    Map<String, byte[]> files;
+    try {
+      files = ZipUtils.unzipFileWithoutDirectories(zipFile.getInputStream());
+    } catch (IllegalArgumentException e) {
+      throw new InvalidStructureInZipFileException(e.getMessage());
+    }
+
+    Set<Dokument> dokumente = findParseAndValidateDokumente(files);
+
+    var manifestationEli = dokumente
+      .stream()
+      .findFirst()
+      .orElseThrow()
+      .getManifestationEli()
+      .asNormEli();
+
+    var binaryFiles = files
+      .entrySet()
+      .stream()
+      .filter(entry ->
+        dokumente
+          .stream()
+          .noneMatch(dokument -> dokument.getManifestationEli().getFileName().equals(entry.getKey())
+          )
+      )
+      .map(entry -> {
+        log.debug("Found binary file: {}", entry.getKey());
+
+        var mimeType = MediaTypeFactory.getMediaType(entry.getKey()).orElseThrow();
+
+        if (SUPPORTED_MIME_TYPES.stream().noneMatch(mimeType::isCompatibleWith)) {
+          throw new UnsupportedFileTypeException(entry.getKey());
+        }
+
+        return new BinaryFile(
+          DokumentManifestationEli.fromNormEli(
+            manifestationEli,
+            entry.getKey().split("\\.")[0],
+            entry.getKey().split("\\.")[1]
+          ),
+          entry.getValue()
+        );
+      })
+      .collect(Collectors.toSet());
+
+    var norm = new Norm(NormPublishState.UNPUBLISHED, dokumente, binaryFiles);
+
+    ldmlDeValidator.validateSchematron(norm);
+
+    if (
+      !norm
+        .getRechtsetzungsdokument()
+        .orElseThrow(MissingRechtsetzungsdokumentException::new)
+        .isVerkuendungsfassung()
+    ) {
+      throw new RechtsetzungsdokumentNotAVerkuendungsfassungException();
+    }
+
+    if (norm.getRegelungstexte().isEmpty() && norm.getBekanntmachungen().isEmpty()) {
+      throw new NoRegelungstextOrBekanntmachungstextException();
+    }
+
+    if (loadNormPort.loadNorm(new LoadNormPort.Command(norm.getWorkEli())).isPresent()) {
+      throw new NormExistsAlreadyException(norm.getWorkEli().toString());
+    }
+
+    log.info(
+      "Verified new norm from import: {} with {} Dokumenten and {} Binary files",
+      norm.getManifestationEli(),
+      norm.getDokumente().size(),
+      norm.getBinaryFiles().size()
+    );
+
+    return norm;
+  }
+
+  private void validateFileIsZipArchive(Resource file) {
+    var mediaType = MediaTypeFactory
+      .getMediaType(file)
+      .orElseThrow(() -> new NotAZipFileException("Invalid filetype: " + file.getFilename()));
+
+    if (!mediaType.includes(MimeType.valueOf("application/zip"))) {
+      throw new NotAZipFileException("Invalid filetype: " + file.getFilename());
+    }
+  }
+
+  private Set<Dokument> findParseAndValidateDokumente(Map<String, byte[]> files) {
+    if (!files.containsKey(RECHTSETZUNGSDOKUMENT_FILENAME)) {
+      throw new MissingRechtsetzungsdokumentException();
```

Comment:
Are you also checking if there is only a single Rechtsetzungsdokument?

Reply:
A rechtsetzungsdokument is always called `rechtsetzungsdokument-1.xml` so there only ever can be one. If there would ever be a file called e.g. `rechtsetzungsdokument-2.xml` I would just say that we treat it like any other additional file

Reply:
Yeah but one AC of the JIRA ticket is:
`MUST: Validate that the archive contains a single rechtssetzungsdokument-1.xml`
Meaning we agreed with PM that we would validate this meaning if there is more than 1 Rechsetzungsdokument we should throw an error right?

Reply:
i interpreted that AC as we should validate that the archive contains a rechtssetzungsdokument (and not 0). It is technically not possible to have to files with that name so there is nothing to validate for the more than 1 case